pallets/capacity/src/types.rs

```diff
@@ -288,31 +292,39 @@ pub trait StakingRewardsProvider<T: Config> {
 	///     - MaxUnlockingChunksExceeded
 	///     - All other conditions that would prevent a reward from being claimed return 'false'
 	fn validate_staking_reward_claim(
-		account_id: AccountId,
-		proof: Hash,
+		account_id: Self::AccountId,
+		proof: Self::Hash,
 		payload: StakingRewardClaim<T>,
 	) -> bool;
-
-	/// Return whether `accountId` can claim a reward. Staking accounts may not claim a reward more than once
-	/// per Era, may not claim rewards before a complete Era has been staked, and may not claim more rewards past
-	/// the number of `MaxUnlockingChunks`.
-	/// Errors:
-	///     - NotAStakingAccount
-	///     - MaxUnlockingChunksExceeded
-	///     - All other conditions that would prevent a reward from being claimed return 'false'
-	fn payout_eligible(account_id: AccountId) -> bool;
 }
 
 /// The information needed to track a Reward Era
 #[derive(
 	PartialEq, Eq, Clone, Default, PartialOrd, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen,
 )]
-pub struct RewardEraInfo<RewardEra, BlockNumber> {
+pub struct RewardEraInfo<RewardEra, BlockNumber>
+where
+	RewardEra: AtLeast32BitUnsigned + EncodeLike,
+{
 	/// the index of this era
-	pub current_era: RewardEra,
+	pub era_index: RewardEra,
```

Comment:
this was renamed since it applies to any stored RewardEraInfo not just the current one. And I just liked "started_at" better, below.